geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -5151,7 +5156,7 @@ private void basicSetPdxSerializer(PdxSerializer serializer) {
   @Override
   public void setReadSerializedForCurrentThread(boolean value) {
     PdxInstanceImpl.setPdxReadSerialized(value);
-    DefaultQuery.setPdxReadSerialized(value);
+    this.getPdxRegistry().setPdxReadSerializedOverride(value);
```

Comment:
In the getPdxReadSerializedByAnyGemFireServices we check to see if pdx registry is null, but here we do not have an equivalent check.  Is there a way that the registry is null when setting?

Reply:
there seems to be a possibility that the registry can be null. Will add the check.